### DISCUSSION TITLE
using with ActiveHash models that store their data in YAML

### DISCUSSION THREAD

**Discussion by drewda** [avo-hq/avo#1691] [Help]
In our web app, we currently have ActiveRecord models for:
- User
- Order (belongs_to a User)

We use the [active_hash](https://github.com/active-hash/active_hash) library to store in YAML files models for:
- Product
- SubscriptionPlan (belongs_to a Product)

Each Order references a Product and references a Subscription Plan.

I'm looking for advice on how to handle these relationships between the ActiveRecord database-backed models and the ActiveHash filed-backed models. Unless I'm missing something, Avo can't handle the ActiveHash models..

For the time being, I just added computed fields like so:

```rb
class OrderResource < Avo::BaseResource
 # ...
  field 'Product', as: :text do |model, resource, view|
    Product.find(model.product_id).name
  rescue
    ''
  end
  field 'Subscription Plan', as: :text do |model, resource, view|
    SubscriptionPlan.find(model.subscription_plan_id).name
  rescue
    ''
  end
  # ...
```

It would be great to be able to use Avo to create and edit these associations (like changing the product on a given order). Any pointers on how to do so?

**Comment by adrianthedev** [comment]
Hey @drewda. Thanks for writing. I'd love to help with that use case.

I haven't used `active_hash` yet. I see that the library has quite a few options and I wouldn't want to try and reproduce the wrong thing.
Would you be open to creating a sample repo with the models and the `yaml` files the way you use them?
You can also add and highlight how you'd like to use them (add the `belongs_to` association and leave a comment next to it where you specify that fails).
You can use [this](https://github.com/avo-hq/avo-reproduction) repo as a starter for that.

Having actual code (and errors), we can both reproduce helps a lot.

Thank you!
